packages/vuetify/src/components/VDataTable/composables/items.ts

```diff
@@ -62,6 +62,11 @@ export function transformItems (
   items: DataTableItemProps['items'],
   columns: InternalDataTableHeader[]
 ): DataTableItem[] {
+  if (isReadonly(items)) {
+    items = items.map(item => toRaw(item))
```

Comment:
This line seems has no impact on prevent reactivity overhead as per testing.

Array map item is not tracked by Vue reactivity system [demo](https://play.vuejs.org/#eNp9Ustu2zAQ/JUFL5IBWYbbogdXNpoGObRAH0h7C3NgpLWsRCIFcukYEPTvXVKW60MR6CByZ2Y5Q+4gbvo+P3oUG1G40jY9gUPy/U7qpuuNJRjA4j6D0nS9J6wyIHOvXmGEvTUdJKxNpJa6NNoRKGthGwTpw6A2sB4z4P+78XExM+Y+N5E579J0AdsdDFJD6JEfVesx71SfqlCPR6ZqwV2ANdqZFvPW1GmyXM4tlsskwpbtWw3fFR1yq3RlunTxSeqRwfAVqykmB+QNYde3ipB3AMVhvbttm/IFnjyR0VAZdDohINvUNdqL22LFzKh4shflMFxFG8czZ2IU54afy9B+KwVnfFg/5hwO1h+lOB/bYbGamEGzujInMkGOg++bOn92RvNzxbuSIhzatGh/9tTwxUixmW4xYKptzeu3WCPrMZvr5QHLl//Un90p1KT4ZdGhPaIUF4yUrZEm+O73Dzzx+gJ2pvIts98A75EfzQePE+2L1xXbvuJFt1/j0DW6/uPuToTazaGC0cAcI18KnrvbN6L/s/s+/xB1PAJi/AupMPsk)

So `item` in line 70 map function isn't reactive by default